src/coreimage.cs

```diff
@@ -330,8 +330,12 @@ public interface CIFilter : NSSecureCoding, NSCopying {
 		[iOS(9,0)]
 		[Static]
 		[Export ("registerFilterName:constructor:classAttributes:")]
+#if XAMCORE_4_0
+		void RegisterFilterName (string name, ICIFilterConstructor constructorObject, NSDictionary<NSString, NSObject> classAttributes);
+#else
```

Comment:
You should add an `[Advice ("The constructorObject argument must implement ICIFilterConstructor")]` here.